### DISCUSSION TITLE
Preserving the picture names

### DISCUSSION THREAD

**Discussion by Sportlots** [ImageMagick/ImageMagick#2955] [Help] [answered]
I'm new to ImageMagick.  I am trying to compress and rotate images in bulk but want to create a new image with the same name in a new directory.  So for example, if I have 25 jpg images in c:\originalImages I want to run a command line to take all of those images, compress and rotate them and put the resulting new compressed and rotated images in a new folder called c:\newImages

I can do what I want without preserving the original filenames, just need a way to make sure I keep the same file names in the new directory.

Is there anyway to do this??

Thanks for any help you can provide.

**Comment by snibgo** [comment]
I suggest you use "mogrify". See https://imagemagick.org/script/mogrify.php

**Comment by fmw42** [comment, marked as answer]
Also see 

https://imagemagick.org/Usage/basics/#mogrify

You will need to create your new directory first. ImageMagick will not create new directories.

Change directories to c:\originalImages

Then for IM 6,

`mogrify -format jpg -path c:\newImages -rotate X -quality Y *.jpg`

If using IM 7,

`magick mogrify -format jpg -path c:\newImages -rotate X -quality Y *.jpg`

Put your rotation angle in place of X and your quality (compression amount in range 0 to 100) in place of Y.

**Comment by Sportlots** [comment]
Thanks, I was briefly testing this and with "mogrify" it seems to ignore my conditional parameter with -rotate.  I use -rotate "90>" because I only want images rotated when currently in landscape mode, but it seems to rotate all images in the folder anyway?

**Reply by fmw42** [reply]
Works fine for me on IM 7.0.10.46 Q16 and also IM 6.9.11.46 Q16 Mac OSX.  What is your IM version and platform?

In IM 6:

```
cd desktop
cd testing
mogrify -format png -path ../test2 -rotate "90>" *

```

In IM7, use magick mogrify

**Comment by Sportlots** [comment]
Using IM7. This is on Windows Server 2012.  When I click on properties of the original file the width is 2988 and length is 5312.

After executing, it rotates the picture.  But if I run the exact same command line, but instead use "90<", then it doesn't rotate the pics.  It is doing the opposite??

**Comment by Sportlots** [comment]
Picture was originally taken on a Samsung S6 phone.  The phone is a few years old.

**Reply by fmw42** [reply]
First, please give the full version. You can find that from 

magick -version

You likely need to auto-orient the image. So in your command line add -auto-orient before the -rotate command.

JPG images often keep a flag that identifies how the camera was turned when taking the picture. -auto-orient corrects the rotation. Some viewers do not take it it into account and so the orientation is wrong.

**Comment by Sportlots** [comment]
I will give that a try and post results here.  The version is: 7.0.10-45  Thanks

**Comment by Sportlots** [comment]
Yes, the auto-orient worked!  Thanks so much.
